Add SPM caching in CI

mokagio opened the pull request:
## Description

What it says in the title 😄 

## Testing instructions
If CI works, we're good.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mokagio committed:
Add SPM caching in CI

spencertransier reviewed (changes requested):
Looks like these scripts would also benefit from adding the SPM cache action: 
- `installable_build.sh` 
- `run-ui-tests.sh`
- `run-unit-tests.sh`

mokagio committed:
Remove no-longer-needed Sentry CLI setup from installable builds script

Must have been missed at the time of
1452014596e51727380342ab1b90b6aaae5cffdb

mokagio commented:
Hey @spencertransier thank you for the review 🙇‍♂️

I just added the call to the installable build step. Great catch, I saw your comment only after noticing it myself while checking for the CI outcome.

As for the `run-...` commands, I did not include them because those step do not build the app so they shouldn't need them. However your comment made me go look at the logs and they do show the packages being fetched:

![image](https://user-images.githubusercontent.com/1218433/236052653-0aa31f28-abb5-467d-9d4b-3fa1c868de2f.png)

I bet this is because of Fastlane `run_tests`, though. As a matter of fact, there is a dedicated `skip_package_dependencies_resolution` option that can be passed to it. I'm going to try that now and see 🤞

mokagio committed:
Skip resolving SPM dependencies in test-without-building step

mokagio committed:
Remove redundant CocoaPods setup from UI tests CI step

This step uses a pre-built binary and therefore does not need to have
dependencies fetched.

mokagio commented:
🤔 looks like something is going [wrong](https://buildkite.com/automattic/woocommerce-ios/builds/12978#_) if resolving the package dependencies is skipped:

```
xcodebuild: error: Could not resolve package dependencies:
  Failed to clone repository https://github.com/krzysztofzablocki/Difference.git:
    Cloning into bare repository '/Users/builder/Library/Developer/Xcode/DerivedData/WooCommerce-btouexuladarzkbinifjxyetlfvd/SourcePackages/repositories/Difference-45c23c1f'...
```

Might need to backtrack or experiment more...

mokagio commented:
🤔 so looks like `xcodebuild -showBuildSettings` does attempt package resolution and for some reason it fails here if we bypass the cache

<img width="1830" alt="Screenshot 2023-05-05 at 4 54 40 pm" src="https://user-images.githubusercontent.com/1218433/236394814-c74dda58-fade-4827-b458-c3c887e9337a.png">

I think for the moment the next best step is to setup the cache in the unit and UI tests as [originally suggested](https://github.com/woocommerce/woocommerce-ios/pull/9623#pullrequestreview-1411686449) by @spencertransier. It'd be good to investigate more later, but I want to prioritize getting the performance improvements.

mokagio committed:
Use SPM cache in all Buildkite steps

I thought we could get away with it in the "test without building"
steps, but evidently that's not the case because CI fails.

See
https://github.com/woocommerce/woocommerce-ios/pull/9623#issuecomment-1535805390

At this point, I prefer the overhead of downloading the cache when it
might be skipped to the additional work of looking into the guts of
Fastlane to see if there's a way to work around the error seen in CI.
Downloading the SPM cache is anyways better that downloading all the
packages during resolution.

mokagio commented on .buildkite/commands/run-ui-tests.sh:
```diff
@@ -20,8 +20,8 @@ tar -xf build-products.tar
 echo "--- :rubygems: Setting up Gems"
 install_gems
 
-echo "--- :cocoapods: Setting up Pods"
-install_cocoapods
```
We can skip the pods because this step doesn't build the app. Unfortunately, the Swift Packages are still necessary because of how Fastlane calls `xcodebuild` under the hood (I think).

spencertransier reviewed (approved):
:shipit:

spencertransier reviewed (approved):
:shipit:
